sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -448,11 +454,19 @@ class SessionCatalog(
    */
   def dropPartitions(
       tableName: TableIdentifier,
-      parts: Seq[TablePartitionSpec],
+      specs: Seq[TablePartitionSpec],
       ignoreIfNotExists: Boolean): Unit = {
+    val tableMetadata = getTableMetadata(tableName)
+    // Verify if each part exactly matches the partition spec of the table definition
+    // Here, we disallow users to drop multiple partition using one spec for atomicity.
+    // For example, the partial spec a=1 could contains multiple partitions a=1/b=1 and a=1/b=2
```

Comment:
this comment is really confusing. I thought the whole point of the patch is to disallow partial specs, e.g. `a=1` in this example?